### PR TITLE
Improve pattern var folder

### PR DESCRIPTION
**Pull request by ice1000**
/cc @imkiva @wsx-ucb Is this correct?

### CODE REVIEW

File: base/src/main/java/org/aya/concrete/visitor/PatternFolder.java [hunk 1]
```diff
@@ -30,15 +31,26 @@
...
+  default @NotNull R fold(@NotNull R acc, @NotNull Arg<Pattern> pat) {
+    return fold(acc, pat.term());
+  }
+
   @MustBeInvokedByOverriders
   default @NotNull R fold(@NotNull R acc, @NotNull Pattern pat) {
     return switch (pat) {
+      case Pattern.List list -> list.elements().foldLeft(acc, this::fold);
+      case Pattern.Tuple tuple -> tuple.patterns().foldLeft(acc, this::fold);
+      case Pattern.BinOpSeq seq -> seq.seq().foldLeft(acc, this::fold);
```
Comment (imkiva) [inline base/src/main/java/org/aya/concrete/visitor/PatternFolder.java:43]: Traverse nested patterns are done in 

https://github.com/aya-prover/aya-dev/blob/723d709514b067719746c0a98f92c4dde1feb3fc/base/src/main/java/org/aya/concrete/visitor/StmtFolder.java#L61-L80

So it is not needed here.

**Review by imkiva — COMMENTED** [review]
This is not correct I suppose.

File: base/src/main/java/org/aya/concrete/visitor/PatternFolder.java (hunk 1 above)
Comment (imkiva) [reply base/src/main/java/org/aya/concrete/visitor/PatternFolder.java:43]: Same for `Expr` and `Stmt`

**Comment by ice1000** [comment]
bors close

**Comment by ice1000** [comment]
Oh I can close it directly LMAO

**Pull request closed without merging**